flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaSink.java

```diff
@@ -132,4 +142,30 @@ public SimpleVersionedSerializer<KafkaWriterState> getWriterStateSerializer() {
     protected Properties getKafkaProducerConfig() {
         return kafkaProducerConfig;
     }
+
+    @Override
+    public LineageVertex getLineageVertex() {
+        // enrich dataset facet with properties
+        if (recordSerializer instanceof KafkaDatasetFacetProvider) {
+            Optional<KafkaDatasetFacet> kafkaDatasetFacet =
+                    ((KafkaDatasetFacetProvider) recordSerializer).getKafkaDatasetFacet();
+
+            if (!kafkaDatasetFacet.isPresent()) {
+                LOG.warn("Provided did not return kafka dataset facet");
+                return null;
```

Comment:
I don't think we are allowed to return null. The interface doesn't specify @Nullable.

Reply:
Nope, interface doesn't specify `@nullable`. We can return `LineageVertex` with empty dataset list instead.